### PR TITLE
fix: default gasLimit with maxGasByBalance

### PR DESCRIPTION
**Pull request by RetricSu**

### CODE REVIEW

File: packages/api-server/src/methods/normalizer.ts [hunk 1]
```diff
@@ -30,10 +30,18 @@ export class EthNormalizer {
...
       "0x" + BigInt(POLY_MAX_BLOCK_GAS_LIMIT).toString(16);
-    const gas = txCallObj.gas || maxBlockGasLimit;
+    const defaultGasLimit =
+      +gasPrice === 0
+        ? maxBlockGasLimit
+        : min(
+            maxBlockGasLimit,
+            await getMaxGasByBalance(this.rpc, fromAddress, gasPrice)
+          );
+    const gas = txCallObj.gas || defaultGasLimit;
```
Comment (keroro520) [inline packages/api-server/src/methods/normalizer.ts:44]:
```suggestion
    let gas = txCallObj.gas;
    if (gas == null) {
        gas = +gasPrice === 0
            ? maxBlockGasLimit
            : min(
                maxBlockGasLimit,
                await getMaxGasByBalance(this.rpc, fromAddress, gasPrice)
              );
    }
```

It is unnecessary to make a `getMaxGasByBalance` call if  `txCallObj.gas` is not `null`

**Review by keroro520 — CHANGES_REQUESTED** [review]

**Review by keroro520 — APPROVED** [review]

File: packages/api-server/src/methods/normalizer.ts [hunk 2]
```diff
@@ -30,10 +30,20 @@ export class EthNormalizer {
...
+    // TODO: use real blockAvailableGas to replace POLY_MAX_BLOCK_GAS_LIMIT
     const maxBlockGasLimit =
       "0x" + BigInt(POLY_MAX_BLOCK_GAS_LIMIT).toString(16);
-    const gas = txCallObj.gas || maxBlockGasLimit;
+    let gas = txCallObj.gas;
+    if (gas == null) {
+      gas =
+        +gasPrice === 0
+          ? maxBlockGasLimit
+          : min(
```
Comment (classicalliu) [inline packages/api-server/src/methods/normalizer.ts:42]: This min can't handle `BigInt`

**Review by classicalliu — CHANGES_REQUESTED** [review]

File: packages/api-server/src/methods/normalizer.ts (hunk 2 above)
Comment (RetricSu) [reply packages/api-server/src/methods/normalizer.ts:42]: there is no `BigInt` type, all use `HexNumber`

Comment (RetricSu) [reply packages/api-server/src/methods/normalizer.ts:42]: oh wait, you mean the overflow problem? I will update the min function

Comment (RetricSu) [reply packages/api-server/src/methods/normalizer.ts:42]: updated in base pr https://github.com/nervosnetwork/godwoken-web3/pull/519/commits/ac4a99e9761110c509e63bcc44ccc8d4b9d0fce8

**Review by classicalliu — APPROVED** [review]

**Pull request merged**